schematic/manifest/generator.py

```diff
@@ -442,7 +452,13 @@ def _add_root_to_component(self, required_metadata_fields):
             # constructor (it's optional) if not, instantiate it
             if not self.additional_metadata:
                 self.additional_metadata = {}
-            self.additional_metadata["Component"] = [self.root]
+            if self.is_file_based:
```

Comment:
Note: All changes in this PR are from running black, with the exception of this line and the next four.